modules/ecs-cluster-ec2-asg/v0.0.1/main.tf

```diff
@@ -0,0 +1,308 @@
+  #checkov:skip=CKV_AWS_355:IAM policy will be set as desired
+  name   = "EcsServiceRolePolicy-${var.cluster_name}-${random_id.code.hex}"
+  role   = aws_iam_role.ecs_service_role.id
+  policy = var.ecs_service_role_policy
+}
+
+# Create Launch Template
+resource "aws_launch_template" "this" {
+  default_version        = 1
+  ebs_optimized          = false
+  name                   = "lt-${var.cluster_name}"
+  image_id               = data.aws_ami.ecs_ami.id
+  instance_type          = var.instance_type
+  key_name               = var.ssh_key_name
+  vpc_security_group_ids = var.security_group_ids
+  user_data              = base64encode(var.user_data != "false" ? var.user_data : local.user_data)
+
+  metadata_options {
+    # Enable Instance Metadata Service V2 
+    http_tokens = "required"
+  }
+
+  iam_instance_profile {
+    name = aws_iam_instance_profile.ecs_instance_profile.id
+  }
+
+  monitoring {
+    enabled = true
+  }
+
+  tag_specifications {
+    resource_type = "network-interface"
+    tags          = var.tags
+  }
+
+  tag_specifications {
+    resource_type = "volume"
+    tags          = var.tags
+  }
+}
+
+# Create Auto-Scaling Group
+resource "aws_autoscaling_group" "this" {
+  name                      = var.cluster_name
+  vpc_zone_identifier       = var.subnet_ids
+  min_size                  = var.min_size
+  max_size                  = var.max_size
+  desired_capacity          = var.desired_capacity
+  health_check_type         = var.health_check_type
+  health_check_grace_period = var.health_check_grace_period
+  default_cooldown          = var.default_cooldown
+  termination_policies      = var.termination_policies
+  protect_from_scale_in     = var.protect_from_scale_in
+  # protect_from_scale_in     = true
+  # To enable managed termination protection for a capacity provider, the Auto Scaling group must have instance protection from scale in enabled.
+
+  launch_template {
+    id      = aws_launch_template.this.id
+    version = "$Latest"
+  }
+
+  tag {
+    key                 = "ecs_cluster"
+    value               = var.cluster_name
+    propagate_at_launch = true
+  }
+
+  dynamic "tag" {
+    for_each = var.tags
+
+    content {
+      key                 = tag.key
+      value               = tag.value
+      propagate_at_launch = true
+    }
+  }
+
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+# Create capacity provider
+# Reference: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/cluster-capacity-providers.html
+resource "aws_ecs_capacity_provider" "this" {
+  #The capacity provider name can have up to 255 characters, including letters (upper and lowercase), numbers, underscores, and hyphens.
+  # The name cannot be prefixed with "aws", "ecs", or "fargate".
+  name = join("-", ["cp", var.cluster_name])
+
+  auto_scaling_group_provider {
+    auto_scaling_group_arn = aws_autoscaling_group.this.arn
+    # managed_termination_protection = "ENABLED"
```

Comment:
why is this commented out?

Reply:
or vice versa: make a variable `managed_termination_protection` and modify the `protect_from_scale_in` parameter based on this variablel

Reply:
This was done for testing the flapping desired count of the ASG

Reply:
Linked `managed_termination_protection` to `protect_from_scale_in`